File: N1.5/N1.5.swift
import Foundation

func stringFixer(line: String) -> (String, Int) {
    var workRow = line
    var count = 0
    var globalCount = 0
    
    for i in 0..<workRow.count-1 {
        let index = workRow.index(workRow.startIndex, offsetBy: i)
        let nextIndex = workRow.index(workRow.startIndex, offsetBy: i+1)
        
        let codeSymbol = Int(workRow[index].asciiValue!)
        let nextSymbol = Character(UnicodeScalar(codeSymbol + 1)!)
        
        if workRow[index] == workRow[nextIndex] {
            count += 1
        }
        
        if count == 2 {
            workRow.insert(nextSymbol, at: nextIndex)
            count = 0
            globalCount += 1
        }
    }
    
    return (workRow, globalCount)
}

let line = "abbcdd"
let (finalRow, globalCount) = stringFixer(line: line)
print("Fixed row - \(finalRow)")
print("Value of changes - \(globalCount)")

File: N1.5/N1.5.cpp
#include <iostream>
#include <string>

using namespace std;

string StringFixer(const string& Line, int& Globalcount) // Func to change string
{
    string WorkRow = Line; 
    int count = 0; 
    for (int i = 0; i < WorkRow.length() - 1; i++)
    {
        int code_symbol = static_cast<int>(WorkRow[i]); 
        code_symbol++; 
        char next_symbol = static_cast<char>(code_symbol); 
        if (WorkRow[i] == WorkRow[i + 1]) 
        {
            count++; 
        }
        if (count == 2) 
        {
            WorkRow.insert(i + 1, 1, next_symbol); 
            count = 0; 
            Globalcount++; 
        }
    }
    return WorkRow; 
}

int main()
{
    int Globalcount = 0; 
    cout << "Enter row, please - ";
    string Line;
    getline(cin, Line); 
    string FinalRow = StringFixer(Line, Globalcount); 
    
    cout << endl << "Fixed row - " << FinalRow << endl << "Value of changes - " << Globalcount;
}